engine/access/rpc/backend/backend_scripts.go

```diff
@@ -108,17 +110,53 @@ func (b *backendScripts) executeScriptOnExecutor(
 	script []byte,
 	arguments [][]byte,
 ) ([]byte, error) {
-	// find few execution nodes which have executed the block earlier and provided an execution receipt for it
-	scriptExecutors, err := b.findScriptExecutors(ctx, blockID)
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "failed to find script executors at blockId %v: %v", blockID.String(), err)
-	}
 	// encode to MD5 as low compute/memory lookup key
 	// CAUTION: cryptographically insecure md5 is used here, but only to de-duplicate logs.
 	// *DO NOT* use this hash for any protocol-related or cryptographic functions.
 	insecureScriptHash := md5.Sum(script) //nolint:gosec
-
 	// try execution on Archive nodes first
+	archiveResult, err := b.executeScriptOnAvailableArchiveNodes(ctx, blockID, script, arguments, insecureScriptHash)
+	// try execution nodes if the script wasn't executed
+	if b.scriptExecValidation {
+		execNodeResult, errExec := b.executeScriptOnAvailableExecutionNodes(
+			ctx, blockID, script, arguments, insecureScriptHash)
+		if bytes.Equal(execNodeResult, archiveResult) && errExec == err {
+			b.logScriptExecutionComparison(blockID, insecureScriptHash,
+				"script execution results on Archive node and EN are equal")
+		} else {
+			b.logScriptExecutionComparison(blockID, insecureScriptHash,
```

Comment:
might be worth adding some debug or trace level logs that print the actual outputs to make debugging this easier